arrow-core-data/src/main/kotlin/arrow/core/Const.kt

```diff
@@ -22,36 +33,266 @@ data class Const<A, out T>(private val value: A) : ConstOf<A, T> {
+      g: Const<A, G>,
+      h: Const<A, H>,
+      i: Const<A, I>,
+      j: Const<A, J>,
+      map: (A, B, C, D, E, F, G, H, I) -> T
+    ): Const<A, T> =
+      b.retag<T>()
+        .combine(SG, c.retag())
+        .combine(SG, d.retag())
+        .combine(SG, e.retag())
+        .combine(SG, f.retag())
+        .combine(SG, g.retag())
+        .combine(SG, h.retag())
+        .combine(SG, i.retag())
+        .combine(SG, j.retag())
+
+    inline fun <A, B, C, D, E, F, G, H, I, J, K, T> mapN(
+      SG: Semigroup<A>,
+      b: Const<A, B>,
+      c: Const<A, C>,
+      d: Const<A, D>,
+      e: Const<A, E>,
+      f: Const<A, F>,
+      g: Const<A, G>,
+      h: Const<A, H>,
+      i: Const<A, I>,
+      j: Const<A, J>,
+      k: Const<A, K>,
+      map: (A, B, C, D, E, F, G, H, I, J) -> T
+    ): Const<A, T> =
+      b.retag<T>()
+        .combine(SG, c.retag())
+        .combine(SG, d.retag())
+        .combine(SG, e.retag())
+        .combine(SG, f.retag())
+        .combine(SG, g.retag())
+        .combine(SG, h.retag())
+        .combine(SG, i.retag())
+        .combine(SG, j.retag())
+        .combine(SG, k.retag())
   }
 
   fun value(): A =
     value
 
+  fun <U> map(f: (T) -> U): Const<A, U> =
+    retag()
+
   fun show(SA: Show<A>): String =
     "$Const(${SA.run { value.show() }})"
 
   override fun toString(): String =
     show(Show.any())
 }
 
+@Deprecated(
+  "Kind is deprecated, and will be removed in 0.13.0. Please use the combine method defined for Const instead",
+  level = DeprecationLevel.WARNING
+)
 fun <A, T> ConstOf<A, T>.combine(SG: Semigroup<A>, that: ConstOf<A, T>): Const<A, T> =
   Const(SG.run { value().combine(that.value()) })
 
+fun <A, T> Const<A, T>.combine(SG: Semigroup<A>, that: Const<A, T>): Const<A, T> =
+  Const(SG.run { value().combine(that.value()) })
+
+@Deprecated(
+  "Kind is deprecated, and will be removed in 0.13.0. Please use the ap method defined for Const instead",
+  ReplaceWith(
+    "Const.mapN(MA, this, arg1)",
+    "arrow.core.Const"
+  ),
+  DeprecationLevel.WARNING
+)
 fun <A, T, U> ConstOf<A, T>.ap(SG: Semigroup<A>, ff: ConstOf<A, (T) -> U>): Const<A, U> =
   fix().retag<U>().combine(SG, ff.fix().retag())
 
+@Deprecated("Kind is deprecated, and will be removed in 0.13.0. Please use one of the provided concrete methods instead")
 fun <T, A, G> ConstOf<A, Kind<G, T>>.sequence(GA: Applicative<G>): Kind<G, Const<A, T>> =
   fix().traverse(GA, ::identity)
 
 inline fun <A> A.const(): Const<A, Nothing> =
   Const(this)
+
+fun <A, T, U> Const<A, T>.contramap(f: (U) -> T): Const<A, U> =
+  retag()
+
+fun <A, T> Const<A, T>.eqv(EQ: Eq<A>, b: Const<A, T>): Boolean =
+  EQ.run {
+    value().eqv(b.value())
+  }
+
+fun <A, T> Eq.Companion.const(EQ: Eq<A>): Eq<Const<A, T>> = object : Eq<Const<A, T>> {
+  override fun Const<A, T>.eqv(b: Const<A, T>): Boolean =
+    eqv(EQ, b)
+}
+
+fun <A, T> Const<A, T>.hashWithSalt(HA: Hash<A>, salt: Int): Int =
+  HA.run {
+    value().hashWithSalt(salt)
+  }
```

Comment:
We're removing `Hash` so these methods don't need to be added. `Hash` can be simply deprecated towards `hashCode()`.